lomond/websocket.py

```diff
@@ -126,6 +128,19 @@ def __exit__(self, exc_type, exc_value, traceback):
         """Close the session (and potentially a socket) on exit."""
         self.session.close()
 
+    def add_header(self, header, value):
+        """Add a user-defined header to the websocket request.
+
+        :param bytes header: Name of the header.
+        :param bytes value: Value of the header.
+
+        """
+        if not isinstance(header, bytes):
```

Comment:
Intentionally not doing any 'helpful' encoding of strs there. It invariably hides bugs that pop up as encoding errors in code you don't understand.